dns.go

```diff
@@ -34,14 +34,25 @@ const (
 
 // From the netmask we can find out the wildcard bits (the bits that are not set in the netmask).
 // This allows us to then calculate the subnets included in the subsequent class block and generate the entries.
-func generateMagicDNSRootDomains(
-	ipPrefix netaddr.IPPrefix,
-) []dnsname.FQDN {
-	// TODO(juanfont): we are not handing out IPv6 addresses yet
-	// and in fact this is Tailscale.com's range (note the fd7a:115c:a1e0: range in the fc00::/7 network)
-	ipv6base := dnsname.FQDN("0.e.1.a.c.5.1.1.a.7.d.f.ip6.arpa.")
-	fqdns := []dnsname.FQDN{ipv6base}
+func generateMagicDNSRootDomains(ipPrefixes []netaddr.IPPrefix) []dnsname.FQDN {
+	fqdns := make([]dnsname.FQDN, 0, len(ipPrefixes))
+	for _, ipPrefix := range ipPrefixes {
+		var generateDnsRoot func(netaddr.IPPrefix) []dnsname.FQDN
+		switch ipPrefix.IP().BitLen() {
+		case 32:
+			generateDnsRoot = generateIPv4DNSRootDomain
 
+		default:
+			panic(fmt.Sprintf("unsupported IP version with address length %d", ipPrefix.IP().BitLen()))
```

Comment:
IPv6 is not supported in MagicDNS?

Reply:
It was stubbed in this commit (which only prepares for multiple namespaces), but IPv6 support was added in a follow-up one.

Reply:
I thought it'd make it easier to review the individual changes this way, but I can squash changes if you'd prefer them that way.

Reply:
Yes, better this way, but maybe mark it with todo or something and tag with you name so we have some pointers to who and when it will be sorted

Reply:
I mean, it is *already* sorted, the comment is on an intermediate commit, not on the top of this branch. :)

Reply:
Are there still concerns about this thread?